sdk/storage/azure-storage-blob/azure/storage/blob/_shared/uploads_async.py

```diff
@@ -421,3 +423,51 @@ async def _upload_chunk(self, chunk_offset, chunk_data):
     # TODO: Implement this method.
     async def _upload_substream_block(self, index, block_stream):
         pass
+
+
+class AsyncIterStreamer():
+    """
+    File-like streaming object for AsyncGenerators.
+    """
+    def __init__(self, generator: AsyncGenerator[Union[bytes, str], None], encoding: str = "UTF-8"):
+        self.generator = generator
+        self.iterator = generator.__aiter__()
+        self.leftover = b""
+        self.encoding = encoding
+
+    def __len__(self):
+        return self.generator.__len__()
```

Comment:
I don't believe generators define `__len__`?

Reply:
Yeah, not sure why this was here. Removed it